src/signer_node.rs

```diff
@@ -663,15 +680,15 @@ impl<T: TapyrusApi, C: ConnectionManager> SignerNode<T, C> {
                     new_signatures.len(),
                     self.params.threshold
                 );
-                if candidate_block.hash().unwrap() != blockhash {
+                if candidate_block.sighash().unwrap() != blockhash {
```

Comment:
Why `Block#hash` returns `Result`? I think it should be return `hash::Hash` directly.

Reply:
Block#sighash can return Hash. I will fix it.

About `Block#hash`, It is possible to return Error because of no proof block.

Reply:
> About Block#hash, It is possible to return Error because of no proof block.

Computing the hash and validating the block should be considered separately. If proof is empty, `Block#hash` should be return hash value with empty proof and another logic checks block validity.

Reply:
I see. I will separate it.

Reply:
Regarding validity, current Block implement is roughly. It should use rust-tapyrus create.

I want to change using rust-tapyrus in other PR sometime soon.

Reply:
Please create new issue.

Reply:
Yes, here #28